index.bs

```diff
@@ -1830,6 +1831,106 @@ To <dfn>get the browsing context info</dfn> given |context|,
 
   1. Return |contexts info|
 
+</div>
+
+<div algorithm>
+To <dfn>await a navigation</dfn> given |context|, |request| and optionally
+|history handling| (default: "<code>default</code>") and |ignore cache| (default: false):
+
+1. Let |navigation id| be the string representation of a
+   [[!RFC4122|UUID]] based on truly random, or pseudo-random numbers.
+
+1. [=Navigate=] |context| with resource |request|, and using |context| as the
+   [=source browsing context=], with navigation id |navigation id|, and history
+   handling |history handling|. If |ignore cache| is true, the navigation must
+   not load resources from the HTTP cache.
+
+   Issue: property specify how the |ignore cache| flag works. This needs to
+   consider whether only the first load of a resource bypasses the cache
+   (i.e. whether this is like initially clearing the cache and proceeding like
+   normal), or whether resources not directly loaded by the HTML parser
+   (e.g. loads initiated by scripts or stylesheets) also bypass the cache.
+
+1.  Let (|event received|, |navigate status|) be [=await=] given
+    «"<code>navigation started</code>", "<code>navigation failed</code>",
+    and "<code>fragment navigated</code>"» and |navigation id|.
```

Comment:
```suggestion
   "<code>fragment navigated</code>"» and |navigation id|.
```

The "and" inside the « » notation doesn't follow https://infra.spec.whatwg.org/#lists.

(And then I noticed this was just moved, not new, so I won't comment on more things in this block.)